File: practice22/12.cpp

//https://www.nowcoder.com/practice/e3d8d4dd9ec740f9b1e7fc1e8574ba21?tpId=40&tqId=21414&tPage=5&rp=5&ru=/ta/kaoyan&qru=/ta/kaoyan/question-ranking
#include<iostream>
#include<string>
using namespace std;

int main()
{
	int m, n, x, y, sum, k;
	while (cin >> m >> n)
	{
		x = 0, y = 0, sum = 0, k = 1;
		for (int i = 0;; i++)
		{
			if (x == m&&y == n)
			{
				cout << sum << endl;
				break;
			}
			sum++;

			if (k == 1)
			{
				x++, y++;
				k++;
				continue;
			}
			if (k == 2)
			{
				x++, y--;
				k++;
				continue;
			}
			if (k == 3)
			{
				x++, y++;
				k++;
				continue;
			}
			if (k==4)
			{
				x--, y++;
				k = 1;
				continue;
			}
		}
	}
	system("pause");
	return 0;
}

File: practice27/12.java


//https://www.nowcoder.com/practice/7037a3d57bbd4336856b8e16a9cafd71?tpId=98&tqId=33012&tPage=10&rp=10&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] pt = new int[n];
        for (int i = 0; i < n; i++) {
            pt[i] = scanner.nextInt();
        }
        int sum = 0;
        for (int i = pt.length - 1; i >= 0; i--) {
            sum= (int) Math.ceil((pt[i]+sum)/2.0);
        }
        System.out.println(sum);
    }
}